fundingmanager.go

```diff
@@ -1437,18 +1443,44 @@ func (f *fundingManager) sendFundingLockedAndAnnounceChannel(
 	}
 	fundingLockedMsg := lnwire.NewFundingLocked(chanID, nextRevocation)
 
+	// If the peer has disconnected before we reach this point, we will need
+	// to wait for him to come back online before sending the fundingLocked
+	// message. This is special for fundingLocked, since failing to send any
+	// of the previous messages in the funding flow just cancels the flow.
+	// But now the funding transaction is confirmed, the channel is open
+	// and we have to make sure the peer gets the fundingLocked message when
+	// it comes back online. This is also crucial during restart of lnd,
+	// where we might try to resend the fundingLocked message before the
+	// server has had the time to connect to the peer.
 	err = f.cfg.SendToPeer(completeChan.IdentityPub, fundingLockedMsg)
 	if err != nil {
-		fndgLog.Errorf("unable to send fundingLocked to peer: %v", err)
-		return
+		fndgLog.Warnf("unable to send fundingLocked to peer: %v. "+
+			"Will retry when online", err)
+
+		connected := make(chan struct{})
```

Comment:
In order to handle a possible flapping case, this should be performed within a loop, something along the lines of: 
```go
for {
     // attempt to send

     // if send succeeds, then break
  
     // re-register the ntfn
}
```